python/util/templates.py

```diff
@@ -110,13 +111,26 @@ def generate_react_components(self):
        newDomain.update(_success, error, baseDomainEmbed);
     }
 
-    function setJob(newJob, success, error) {
-       function _success () {
-         window.job = newJob;
-         redrawAll();
-         success();
-       }
-       newJob.update(_success, error, baseJobEmbed);
+    function updateJob(options) {
+        var jobCopy = new MERCHI.Job();
+        var embed = options.embed ? options.embed : baseJobEmbed;
+        function _success (jobData) {
+            window.job = jobData
+            redrawAll();
+            options.success();
+        }
+        jobCopy.id(window.job.id());
+        job.get(_success, options.error, embed);
+    }
+
+    function setJob(options) {
+        var embed = options.embed ? options.embed : baseJobEmbed;
```

Comment:
no, you cannot use an embed supplied only by one component, you must **extend** or **merge** the base embed, otherwise other components are interfered with by not having the base properties available.